src/aspire/source/image.py

```diff
@@ -425,6 +471,7 @@ def _apply_source_filters(self, im_orig, indices):
             self.filter_indices[indices],
         )
 
+    @as_copy
```

Comment:
Just curious, was this one needed for anything other than `ArrayImageSource`?

Reply:
I do see some uses of `cache()` for `RelionSource` objects when executing certain commands. For example:
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/e3145c1a3c5a9b4a18d3c917e15e20d4569d8088/src/aspire/commands/cov3d.py#L51
https://github.com/ComputationalCryoEM/ASPIRE-Python/blob/e3145c1a3c5a9b4a18d3c917e15e20d4569d8088/src/aspire/commands/denoise.py#L81

I don't think `RelionSource` is related to `ArrayImageSource`. Or perhaps I'm misunderstanding the question..

Reply:
I don't know that the `cache` needs to make/return a copy. The image array it is storing should be immutable.  Cache is just saying, generate and store stuff inside self now, so it is ready.  In other words, `cache` can probably return itself after making and assigning the cached data array.  I don't personally consider it preprocessing (it isn't mutating anything in the pipeline).

I guess there is an argument that decorating this with copy is safer, and there might be some subtlety I am missing.

The behavior for `ArrayImageSource` is slightly different, because it is already cached (backed by an array already).

Reply:
The only scenario I can think of is if one wants to compare the performance of an uncached `src` vs. a cached version of it, without having to recreate it.

I don't know the typical use case of these methods, so happy to take it off if you think (of course feel free to tweak this branch directly).